src/steamship/agents/llms/openai.py

```diff
@@ -119,7 +116,31 @@ def chat(self, messages: List[Block], tools: Optional[List[Tool]], **kwargs) ->
 
         logging.info(f"OpenAI ChatComplete ({messages[-1].as_llm_input()})", extra=extra)
 
-        tool_selection_task = self.generator.generate(input_file_id=temp_file.id, options=options)
-        tool_selection_task.wait()
-
-        return tool_selection_task.output.blocks
+        # for streaming use cases, we want to always use the existing file
+        # the way to detect this would be if all messages were from the same file
+        if self._from_same_file(blocks=messages):
+            file_id = messages[0].id
+            block_ids = [b.id for b in messages]
+            generate_task = self.generator.generate(
+                input_file_id=file_id,
+                input_file_block_index_list=block_ids,
+                options=options,
+                append_output_to_file=True,
+            )
```

Comment:
If we're going to generate with streaming _blocks_, we also need to pass `streaming=True` here.  That will result in the generate_task.output.blocks being allocated but not yet completed.  This means that the caller then has to be aware of whether or not they're streaming.

See: https://github.com/steamship-core/python-client/pull/548

Reply:
Walking through this together is probably the best way to answer the confusion here. I _think_ we want `streaming=True` to *always* be the case really, but I don't think the caller *here* cares, because there is a `wait()` on this task anyway. It is more that the downstream consumer that has access to the `File` cares, right?